pandas/tests/indexes/numeric/test_numeric.py

```diff
@@ -347,8 +347,12 @@ def test_constructor(self, dtype):
         val = arr[0] + 3000
 
         # this should not change index
-        arr[0] = val
-        assert new_index[0] != val
+        if not using_copy_on_write:
+            arr[0] = val
+            assert new_index[0] != val
```

Comment:
Similarly here, we want to test that the `copy=True` keyword is honored (I assume, based on the comment). So maybe add a copy() to the line creating `arr` above: `arr = index.values.copy()`. Then `arr` is writeable, and the test can be done as normal (ensuring that `arr` was actually copied when creating the Index.

Reply:
makes sense